packages/sw-cache-expiration/demo/service-worker.js

```diff
@@ -1,26 +1,33 @@
 /* eslint-env worker, serviceworker */
 /* global goog */
 
-importScripts('../build/routing.js',
-  '../../sw-runtime-caching/build/runtime-caching.js');
+importScripts(
+  '../../sw-routing/build/sw-routing.min.js',
+  '../../sw-runtime-caching/build/sw-runtime-caching.min.js',
+  '../build/sw-cache-expiration.min.js'
+);
 
 // Have the service worker take control as soon as possible.
-self.addEventListener('install', (event) => {
-  self.skipWaiting();
-  event.waitUntil(
-    caches.open(goog.runtimeCaching.defaultCacheName)
-      .then((cache) => cache.add('offline.html'))
-  );
-});
+self.addEventListener('install', () => self.skipWaiting());
 self.addEventListener('activate', () => self.clients.claim());
 
-const routes = [
-  new goog.routing.Route({
-    when: ({event}) => event.request.mode === 'navigate',
-    handler: ({event}) => {
-      return fetch(event.request).catch(() => caches.match('offline.html'));
-    },
-  }),
-];
+const cacheName = 'text-files';
+const requestWrapper = new goog.runtimeCaching.RequestWrapper({
+  cacheName,
+  behaviors: [
+    new goog.cacheExpiration.Behavior({
+      cacheName,
+      maxEntries: 2,
+      maxAgeSeconds: 10,
+    }),
+  ],
+});
+
+const route = new goog.routing.RegExpRoute({
+  regExp: /\.txt$/,
+  handler: new goog.runtimeCaching.StaleWhileRevalidate({requestWrapper}),
```

Comment:
This level of indirection makes it hard to reason with what is going on.

StaleWhileRevalidate > RequestWrapper > CacheExpirationBehavior.

Nothing to action on here, but reading this code, I have no idea why the RequestWrapper exists and how it works with the StaleWhileRevalidate and then how that plays with Behavior :(